File: src/VRE.Vridge.API.Client/Messages/v1/Control/Responses/EndpointStatus.cs
namespace VRE.Vridge.API.Client.Messages.v1.Control.Responses
{    
    public class EndpointStatus : ControlResponseHeader
    {
        public string Name;

        public EndpointStatus(string name)
        {
            this.Code = (int) ControlResponseCode.OK;
            this.Name = name;
        }
    }
}

File: src/java/VRE.Vridge.API/VRE.Vridge.API.Client/src/main/java/com/riftcat/vridge/api/client/java/control/responses/EndpointStatus.java
package com.riftcat.vridge.api.client.java.control.responses;

import com.riftcat.vridge.api.client.java.control.ControlResponseCode;

public class EndpointStatus extends ControlResponseHeader {

    public String Name;
    public String InUseBy;

    public EndpointStatus(String endpointName){
        this.Code = ControlResponseCode.OK;
        this.Name = endpointName;
    }

    public String codeString(){
        if(Code == ControlResponseCode.NotAvailable){
            return "n/a";
        }
        if(Code == ControlResponseCode.InUse){
            return "In use by " + InUseBy;
        }
        if(Code == ControlResponseCode.OK){
            return "Ready";
        }

        return "unknown";
    }

}
